Streetcode/Streetcode.BLL/Services/Authentication/TokenService.cs

```diff
@@ -0,0 +1,122 @@
+﻿using System.Data;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Text;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Localization;
+using Microsoft.IdentityModel.Tokens;
+using Streetcode.BLL.Interfaces.Users;
+using Streetcode.DAL.Entities.Users;
+using Streetcode.DAL.Persistence;
+
+namespace Streetcode.BLL.Services.Authentication
+{
+    public sealed class TokenService : ITokenService
+    {
+        private readonly SigningCredentials _signingCredentials;
+        private readonly string _jwtIssuer;
+        private readonly string _jwtAudience;
+        private readonly string _jwtKey;
+        private readonly IStringLocalizer<TokenService> _stringLocalizer;
+        private readonly StreetcodeDbContext _dbContext;
+        private readonly JwtSecurityTokenHandler _jwtSecurityTokenHandler;
+
+        public TokenService(IConfiguration configuration, IStringLocalizer<TokenService> stringLocalizer, StreetcodeDbContext dbContext)
+        {
+            _jwtIssuer = configuration["Jwt:Issuer"] !;
+            _jwtAudience = configuration["Jwt:Audience"] !;
+            _jwtKey = configuration["Jwt:Key"] !;
+
+            _stringLocalizer = stringLocalizer;
+            _dbContext = dbContext;
+
+            var symmetricSecurityKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(_jwtKey));
+            _signingCredentials = new SigningCredentials(symmetricSecurityKey, SecurityAlgorithms.HmacSha256);
+            _jwtSecurityTokenHandler = new JwtSecurityTokenHandler();
+        }
+
+        public JwtSecurityToken GenerateJWTToken(User? user)
+        {
+            // Throw exception if argument is null.
+            if (user is null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+
+            // Get roles of user.
+            var userRoleId = _dbContext.UserRoles
+                .AsNoTracking()
+                .Where(userRole => userRole.UserId == user.Id)
+                .Select(userRole => userRole.RoleId)
+                .FirstOrDefault();
+            var userRoleName = _dbContext.Roles
```

Comment:
It's better to fetch userRole and in the next line `var userRoleName = userRole!.Name`
Because here in the chain you are making request AND retrieving specific data from the result, it's less readable